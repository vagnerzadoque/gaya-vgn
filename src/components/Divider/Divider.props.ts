export type DividerVariants = 'full-bleed' | 'inset' | 'middle'

export interface DividerProps {
  /**
   * Specify an optional className to be added to your Divider
   */
  className?: string

  /**
   * Optional ID for testing
   */
  testID?: string;

  /**
   * The variant to use.
   */
  variant?: DividerVariants;
}
