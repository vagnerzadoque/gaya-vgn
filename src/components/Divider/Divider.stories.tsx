import React from 'react'
import type { Meta, StoryObj } from '@storybook/react'
import { Divider, DividerProps } from '.'

const componentStatus = `
> A divider is a thin line that groups content in lists and layouts.

## Properties
| Property             | Values                       |    Status           |
|---                   |                           ---|                  ---|
| **variant**          | full-bleed, inset, middle    | âœ… Available        |
## Technical Usages Examples
`

const Template: React.FC = ({ children }) => (
  <div style={{ maxWidth: 200, border: '1px solid #00000014', padding: '16px 0' }}>
    {children}
  </div>
)

const meta: Meta<DividerProps> = {
  title: 'Components/Divider',
  component: Divider,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
};

export default meta;
type Story = StoryObj<DividerProps>;

export const Playground: Story = {
  render: (args) => (
    <Template>
      <Divider {...args} />
    </Template>
  )
};

export const FullBleed: Story = {
  args: {
    variant: 'full-bleed'
  },
  render: (args) => (
    <Template>
      <Divider {...args} />
    </Template>
  )
};

export const Inset: Story = {
  args: {
    variant: 'inset'
  },
  render: (args) => (
    <Template>
      <Divider {...args} />
    </Template>
  )
};

export const Middle: Story = {
  args: {
    variant: 'middle'
  },
  render: (args) => (
    <Template>
      <Divider {...args} />
    </Template>
  )
};
