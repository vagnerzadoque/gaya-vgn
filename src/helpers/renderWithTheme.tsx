import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import themes from '@naturacosmeticos/natds-themes'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component: JSX.Element) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider theme={themes.natura.light}>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
