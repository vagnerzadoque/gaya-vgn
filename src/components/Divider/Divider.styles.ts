import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { DividerProps } from './Divider.props'

type DividerStyleProps = Required<Pick<DividerProps, 'variant'>>

const styles = createUseStyles((theme: Theme) => ({
  divider: {
    border: 0,
    height: 1,
    backgroundColor: theme.color.lowEmphasis,
    marginTop: 0,
    marginBottom: 0,
    marginRight: ({ variant }: DividerStyleProps) => (variant === 'middle' ? theme.spacing.standard : 0),
    marginLeft: ({ variant }: DividerStyleProps) => (variant === 'full-bleed' ? 0 : theme.spacing.standard)
  }
}))

export default styles
