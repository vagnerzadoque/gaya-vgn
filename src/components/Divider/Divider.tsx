import React from 'react'
import { DividerProps } from './Divider.props'
import styles from './Divider.styles'

const Divider = React.forwardRef<HTMLHRElement, DividerProps>(
  ({
    className = '',
    testID,
    variant = 'full-bleed',
    ...props
  }, ref) => {
    const { divider } = styles({ variant })

    return (
      <hr
        className={`${className} ${divider} Divider-gaya`}
        data-testid={testID}
        ref={ref}
        {...props}
      />
    )
  }
)

export default Divider
