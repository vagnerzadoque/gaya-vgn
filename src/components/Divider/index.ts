import Divider from './Divider'

export { Divider }
export { DividerProps } from './Divider.props'
export default Divider
