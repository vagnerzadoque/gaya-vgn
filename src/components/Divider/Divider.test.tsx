import * as React from 'react'
import Divider from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('Divider component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Divider />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is inset', () => {
    const { styles, component } = renderWithTheme(<Divider variant="inset" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is middle', () => {
    const { styles, component } = renderWithTheme(<Divider variant="middle" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
