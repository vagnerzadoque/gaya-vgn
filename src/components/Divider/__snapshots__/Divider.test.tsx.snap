// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component should render correctly when the variant is inset 1`] = `
[
  ".divider-0-2-1 {
  border: 0;
  height: 1px;
  margin-top: 0;
  margin-bottom: 0;
  background-color: #8c8c8c;
}
.divider-d0-0-2-2 {
  margin-right: 0;
  margin-left: 24px;
}",
  <div>
    <hr
      class=" divider-0-2-1 divider-d0-0-2-2 Divider-gaya"
    />
  </div>,
]
`;

exports[`Divider component should render correctly when the variant is middle 1`] = `
[
  ".divider-0-2-1 {
  border: 0;
  height: 1px;
  margin-top: 0;
  margin-bottom: 0;
  background-color: #8c8c8c;
}
.divider-d0-0-2-2 {
  margin-right: 24px;
  margin-left: 24px;
}",
  <div>
    <hr
      class=" divider-0-2-1 divider-d0-0-2-2 Divider-gaya"
    />
  </div>,
]
`;

exports[`Divider component should render correctly with default props 1`] = `
[
  ".divider-0-2-1 {
  border: 0;
  height: 1px;
  margin-top: 0;
  margin-bottom: 0;
  background-color: #8c8c8c;
}
.divider-d0-0-2-2 {
  margin-right: 0;
  margin-left: 0;
}",
  <div>
    <hr
      class=" divider-0-2-1 divider-d0-0-2-2 Divider-gaya"
    />
  </div>,
]
`;
